### PR TITLE
Add search result pagination

### PR DESCRIPTION
**Pull request by fzxu**
this is for issue: https://github.com/intuit/katlas/issues/24

* do not fire action when typing
* pagination for search result
* add search input in the page (remove from header menu bar) as the QSL is much longer than text search
* disabled some test form home page due to no action will be fired when typing
* some format as usual

### CODE REVIEW

**Comment by fzxu** [comment]
https://github.com/intuit/katlas/issues/24

File: app/src/services/ApiService.js [hunk 1]
```diff
@@ -16,38 +18,74 @@ export default class ApiService {
...
-    return requestHelper(ALL_SERVICE_URL + ALL_SERVICE_CONTEXT +
-      ENTITY_SERVICE_PATH + uid);
+    return requestHelper(
+      getServiceURL() + ALL_SERVICE_CONTEXT + ENTITY_SERVICE_PATH + uid
+    );
   }
 }
 
+const getServiceURL = () => {
+  return process.env.NODE_ENV === 'test'
```
Comment (davemasselink) [inline app/src/services/ApiService.js]: @fzxu, might it be easier simply for the `KATLAS_API_URL` value to equal localhost when run outside of a deployed env?

File: app/src/services/ApiService.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    // verify the url requested by the api service matches the expected url mocked.
+    fetchMock.get('http://localhost/v1/qsl/Cluster[$$first=25,offset=0]{*}', response);
+
+    ApiService.getQSLResult('Cluster{*}', 0, 25);
+
+    // verify the url requested by the api service matches the expected url mocked.
+    fetchMock.get('http://localhost/v1/qsl/Cluster[@name="abc"$$first=50,offset=100]{*}', response);
+
+    ApiService.getQSLResult('Cluster[@name="abc"{*}', 2, 50);
```
Comment (davemasselink) [inline app/src/services/ApiService.test.js]: is this request meant to include an invalid query? if not, I believe we'd want a closing square bracket `]` after `"abc"` in order to make it fully valid

File: app/src/actions/queryActions.js [hunk 3]
```diff
@@ -3,71 +3,90 @@ import * as notifyActions from './notifyActions';
...
-
-export const changeQuery = str => ({
-  type: types.CHANGE_QUERY,
-  query: str
-});
-
-const submitQueryAction = () => ({
-  type: types.SUBMIT_QUERY,
-});
+const QSL_TAG = '{';
```
[the remark this answers is not in the document]
Comment (fzxu) [reply app/src/actions/queryActions.js]: `@` is optional for QSL query vs `{` is mandatory. but I agree we should use the same regex which used for parsing to also set the query type.

File: app/src/actions/queryActions.test.js [hunk 4]
```diff
@@ -57,7 +48,7 @@ describe('asynch query actions', () => {
       Promise.resolve(new Response(JSON.stringify(MOCK_RESP_EMPTY), {status:200})));
     return store.dispatch(actions.fetchQuery('doesntMatter')).then(() => {
       //now we're looking for the 2nd action to have been triggered
```
[the remark this answers is not in the document]
Comment (fzxu) [reply app/src/actions/queryActions.test.js]: added the comments

**Pull request merged**